conbench/api/benchmarks.py

```diff
@@ -182,14 +186,75 @@ def post(self) -> None:
         tags:
           - Benchmarks
         """
-        # Here it sould be easy to make `data` have a precise type (that mypy
+        # Here it should be easy to make `data` have a precise type (that mypy
         # can use) based on the schema that we validate against.
         data = self.validate_benchmark(self.schema.create)
+
+        # This next block inspects and mutates `tags`, with the goal that all
+        # keys are non-empty strings, and all values are non-empty strings.
+        # ideally this should be done as part of schema validation above.
+        tags = data["tags"]
+        for key, value in list(tags.items()):
+            # In JSON, a key is always of type string. We can rely on this
+            # here, but codify this invariant.
+            assert isinstance(key, str)
```

Comment:
We have tags which are not strings, e.g. [query ID and scale factor for TPC-H](https://github.com/voltrondata-labs/benchmarks/blob/75bc80e1bc75b428734f7a018a7f5460d6094274/benchmarks/tpch_benchmark.py#L8-L9). They can be coerced to strings (if that's desirable?), but if we merge this before doing so we're going to get a lot of rejected results.

Reply:
Oops, query ID does get formatted as a string. But `memory_map` is a bool: https://github.com/voltrondata-labs/benchmarks/blob/75bc80e1bc75b428734f7a018a7f5460d6094274/benchmarks/tpch_benchmark.py#L26-L27